boards/common/blxxxpill/Makefile.include

```diff
@@ -7,6 +7,23 @@ ifeq (dfu-util,$(PROGRAMMER))
 
   # Leave some space at the beginning of the flash for the bootloader
   ROM_OFFSET ?= 0x2000
+
+  # If using STM32duino bootloader, this change is necessary.
+  # Bootloader available at
+  # github.com/rogerclarkmelbourne/STM32duino-bootloader/tree/master/binaries
+  ifeq (stm32duino,$(BOOTLOADER))
+    FFLAGS_OPTS ?= -R
+    FFLAGS ?= --device $(DFU_USB_ID) \
+              --alt $(DFU_ALT) \
+              --download $(FLASHFILE) \
+              $(FFLAGS_OPTS)
+
+    # Flashing may be easier when uncommenting the "RESET_STM" line and
+    # using the binary "upload-reset" from rogerclarkmelbourne/Arduino_STM32.
+    # github.com/rogerclarkmelbourne/Arduino_STM32/tree/master/tools/linux/upload-reset
+    # RESET_STM := $(shell $(RIOTBOARD)/common/blxxxpill/bin/upload-reset /dev/ttyACM0 750)
```

Comment:
Btw: How does this affect flashing?

Reply:
That`s is why I did not incorporate this code into the commit, just mentioned the possibility.
Indeed, there are some caveats:

1- Arduino's structure uses USB for its stdio (which should be one of my priorities soon). Hence, this `RESET_STM` works great (if CPU is not in `delay()`). But in Riot, USB is not active after bootloading, so the reset won't kick in after that, reset must be pressed manually.

2- During the flash of the Riot's default dfu configuration, it does not wait the device to be reset, then it makes very hard to synchronize both actions manually. With this `RESET_STM`, stm32duino receives a command to wait for flash and makes dfu synchronization easier (that was my impression). So I decided to leave this suggestion commented.

I wouldn't mind removing this, but what is the problem of some more information/help?

Reply:
btw.: this seems very similar to what is done on sam0 with the Arduino bootloader (#12304), so the RIOT USB stack already has a baudrate hack in place to trigger a bootloader reset.

Unfortunately the USB peripheral is not the same across all stm32, so currently stm32f1 has no USB support in RIOT.

Reply:
In any case, this reset is not absolutely necessary, and it may not always work, because of USB peripheral not available.
I wonder how easy would it be to port the code from Arduino, they use USB as STDIO.

Reply:
You can use USB as STDIO too in RIOT, just not on stm32f1 (yet) - you'd have to implement/adopt [`stm32_common/periph/usbdev.c`](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32_common/periph/usbdev.c).

On stm32f4 (e.g. #12778) stdio over USB works just fine.

Reply:
I'm sorry, I meant that I agree with you.
My point is that in Arduino `stm32f1` is already working with USB as STDIO too, so maybe just observing their code and adapting into RIOT could be possible/faster.

Reply:
I don't think there is any technical reason for why it's not done for stm32f1 yet - it's just that nobody has done it yet/is actively working on it.

If you want to work on this, that would be great!

Reply:
I should. I would like to use it too, but first I wanna get this one done.